api/v1alpha1/instaslice_types.go

```diff
@@ -95,6 +95,8 @@ type AllocationDetails struct {
 	// memory represents amount of memory requested by user workload
 	// +required
 	Memory int64 `json:"memory"`
+	// to prevent double metric incrementation
+	IsMetricProcessed bool `json:"isMetricProcessed,omitempty"`
```

Comment:
This needs to be on the status object. A variable like this doesn't seem like the right fit. Have you considered the ObservedGeneration on the Status object?

Reply:
You are right, updated